packages/sdk/src/accountTransactions.ts

```diff
@@ -259,8 +259,25 @@ export class InitContractHandler implements AccountTransactionHandler<InitContra
         return Buffer.concat([serializedAmount, serializedModuleRef, serializedInitName, serializedParameters]);
     }
 
-    deserialize(): InitContractPayload {
-        throw new Error('deserialize not supported');
+    deserialize(serializePayload:Cursor): InitContractPayload {
+        const amount = serializePayload.read(8).readBigUInt64BE(0);
+        const moduleRef = serializePayload.read(32);
+        
+        const initNameLength = serializePayload.read(2).readUInt16BE(0);
+        const initName = serializePayload.read(initNameLength);
+        const initNameAfterConversion = fromString(initName.toString('utf8'));
+
+        const paramLength = serializePayload.read(2).readUInt16BE(0);
+        const param = serializePayload.read(paramLength);
+        const paramBuffer = Parameter.fromBuffer(param.buffer);
+
+        return {            
+           amount: CcdAmount.fromMicroCcd(amount),
+           moduleRef: ModuleReference.fromBuffer(moduleRef),
+           initName: ContractName.fromInitName(initNameAfterConversion), 
+           param: paramBuffer,
+           maxContractExecutionEnergy: Energy.create(0n), //setting this to zero, bluepaper does not specify this
```

Comment:
I would instead describe this as

```suggestion
           // The execution energy cannot be recovered, as it is not part of the payload serialization.
           maxContractExecutionEnergy: Energy.create(0n), 
```

Reply:
changing the comment